Sources/StreamChat/Database/DTOs/MessageDTO_Tests.swift

```diff
@@ -1038,6 +1038,91 @@ class MessageDTO_Tests: XCTestCase {
+        // Here, secondMessage and thirdMessage quote the firstMessage
+        
+        let firstMessageId: MessageId = .unique
+        let secondMessageId: MessageId = .unique
+        let thirdMessageId: MessageId = .unique
+        let currentUserId: UserId = .unique
+        let messageAuthorId: UserId = .unique
+        let channelId: ChannelId = .unique
+        
+        try database.createCurrentUser(id: currentUserId)
+        try database.createChannel(cid: channelId, withMessages: false)
+        
+        var createdMessages: [MessagePayload<NoExtraData>] = []
+        
+        let messageIdToQuotedIdMapping = [
+            secondMessageId: firstMessageId,
+            thirdMessageId: firstMessageId
+        ]
+        
+        let messageToBeQuoted: MessagePayload<NoExtraData> = .dummy(
+            messageId: firstMessageId,
+            quotedMessage: nil,
+            attachments: [],
+            authorUserId: messageAuthorId,
+            latestReactions: [],
+            ownReactions: [],
+            channel: .dummy(cid: channelId),
+            pinned: true,
+            pinnedByUserId: .unique,
+            pinnedAt: .unique,
+            pinExpires: .unique
+        )
+        createdMessages.append(messageToBeQuoted)
+        
+        messageIdToQuotedIdMapping.forEach { (messageId, quotedMessageId) in
+            let message: MessagePayload<NoExtraData> = .dummy(
+                messageId: messageId,
+                quotedMessage: .dummy(
+                    messageId: quotedMessageId,
+                    authorUserId: messageAuthorId
+                ),
+                attachments: [],
+                authorUserId: messageAuthorId,
+                latestReactions: [],
+                ownReactions: [],
+                channel: .dummy(cid: channelId),
+                pinned: true,
+                pinnedByUserId: .unique,
+                pinnedAt: .unique,
+                pinExpires: .unique
+            )
+            createdMessages.append(message)
+        }
+                
+        try createdMessages.forEach { messagePayload in
+            try database.writeSynchronously { session in
+                // Save the message
+                try session.saveMessage(payload: messagePayload, for: channelId)
+            }
+        }
+
+        var loadedMessages: [ChatMessage] = []
+        try [firstMessageId, secondMessageId, thirdMessageId].forEach { messageId in
+            // Load the messages one by one from the db and save them
+            let loadedMessage: ChatMessage = try XCTUnwrap(
+                database.viewContext.message(id: messageId)?.asModel()
+            )
+            loadedMessages.append(loadedMessage)
+        }
+        
+        XCTAssertEqual(createdMessages.count, loadedMessages.count)
+        XCTAssertEqual(loadedMessages.first?.quotedMessage, nil) // The very first message doesn't quote any message
```

Comment:
nit: in tests we usually put comments above
```suggestion
        // The very first message doesn't quote any message
        XCTAssertEqual(loadedMessages.first?.quotedMessage, nil)
```

Reply:
Changed to match our code style